src/server/controllers/gitController.ts

```diff
@@ -0,0 +1,44 @@
+export {};
+
+const exec = require("child_process").execFile;
+
+import { Request, Response, NextFunction } from "express";
+
+const gitController: any = {};
+
+function execShellCommand(shellCommand: string, args: Array<string>) {
+  return new Promise((resolve, reject) => {
+    exec(
+      shellCommand,
+      args,
+      (error: string, stdout: string, stderr: string) => {
+        if (error) {
+          console.warn(error);
+        }
+        console.warn(stderr);
+        resolve(stdout ? stdout : stderr);
+      }
+    );
+  });
+}
+
+gitController.cloneRepo = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const { username, repos } = res.locals;
+
+  // TODO: add repoName from res.locals
+  const repoName = repos[1].name;
```

Comment:
`repos[1]` is hard-coded for testing purposes. This will need to be changed once the front-end sends information about which repositories should be cloned. The full flow should use an array of repositories that will be looped through and cloned by this middleware.